compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -281,15 +291,17 @@ template <typename TileSizesType>
 static Value buildReductionStrategyBlockDistributionPart(
     ImplicitLocOpBuilder &b, Value variantH, Value originalFillH,
     Value reductionH, Value optionalFusionRootH,
-    TileSizesType tileSizes0Generic, bool hasLeadingEltwise = false) {
+    TileSizesType tileSizes0Generic, bool hasLeadingEltwise = false,
+    bool hasTrailingEltwise = false) {
   // Step 1. Split the reduction to get meatier parallelism.
   // TODO: use a scf.foreach_thread for this.
   auto splitReductionTransformOp =
       b.create<SplitReductionOp>(reductionH,
                                  /*splitFactor=*/2,
                                  /*insertSplitDimension=*/1);
-  ReductionSplitResult rs = buildExpansionBubbleUp(
-      b, variantH, splitReductionTransformOp, hasLeadingEltwise);
+  ReductionSplitResult rs =
+      buildExpansionBubbleUp(b, variantH, splitReductionTransformOp,
```

Comment:
2nd commit : Since `ApplyPatternsOp` uses the tracking listener I wonder whether we can use that fact to avoid rematching and simplifying the magic bools re `hasLeading/hasTrailing` in the API (and instead pass `FailureOr<Value>` matched earlier + tracked) ?

Not for this PR / stack and I forget the deeper details of nested handles invalidation but it seems doable ?